project_task_parent_due_auto/models/project_task.py

```diff
@@ -0,0 +1,36 @@
+from datetime import date
+
+from odoo import api, models
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    @api.model
+    def create(self, values):
+        tasks = super().create(values)
+        if "date_deadline" in values:
+            tasks.update_parent_task_dates()
+        return tasks
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "date_deadline" in vals:
+            self.update_parent_task_dates()
+        return res
+
+    def update_parent_task_dates(self):
+        for task in self:
+            parent = task.parent_id
+            if parent:
+                if len(parent.child_ids) == 1:
+                    parent.date_deadline = task.date_deadline
+                else:
+                    closest_task = min(
+                        parent.child_ids, key=lambda t: t.date_deadline or date.max
+                    )
+                    if (
+                        closest_task.date_deadline
+                        and parent.date_deadline != closest_task.date_deadline
+                    ):
+                        parent.date_deadline = closest_task.date_deadline
```

Comment:
Please don't over-complicate:

```suggestion
                deadline = min(parent.child_ids.mapped("date_deadline"))
                if deadline and parent.date_deadline != deadline
                    parent.date_deadline = deadline
```